common/src/main/java/org/jboss/hal/testsuite/creaper/command/BackupAndRestoreAttributes.java

```diff
@@ -0,0 +1,163 @@
+package org.jboss.hal.testsuite.creaper.command;
+
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.Property;
+import org.wildfly.extras.creaper.core.CommandFailedException;
+import org.wildfly.extras.creaper.core.online.OnlineCommand;
+import org.wildfly.extras.creaper.core.online.OnlineCommandContext;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.Batch;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.ReadResourceOption;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * @author Jan Kasik <jkasik@redhat.com>
+ *         Created on 27.11.15.
+ */
+public class BackupAndRestoreAttributes {
+
+    private final Address address;
+    private final Map<String, String> dependencies;
+    private final Set<String> excluded;
+
+    private ModelNode backup;
+
+    private BackupAndRestoreAttributes(Builder builder) {
+        this.address = builder.address;
+        this.dependencies = builder.dependencies;
+        this.excluded = builder.excluded;
+    }
+
+    private final OnlineCommand backupPart = new OnlineCommand() {
+
+        @Override
+        public void apply(OnlineCommandContext ctx) throws Exception {
+            if (BackupAndRestoreAttributes.this.backup != null) {
+                throw new CommandFailedException("Backup has been already made!");
+            }
+            Operations ops = new Operations(ctx.client);
+
+            BackupAndRestoreAttributes.this.backup = ops.readResource(address, ReadResourceOption.INCLUDE_DEFAULTS, ReadResourceOption.ATTRIBUTES_ONLY).value();
+        }
+    };
+
+    private final OnlineCommand restorePart = new OnlineCommand() {
+
+        //<dependency which needs to be processed before dependent attribute is added to batch, dependent attributes>
+        private final Map<String, Set<Property>> waitingForDependency = new HashMap<>(); //waiting queue
+        private final Set<String> processedAttributes = new HashSet<>(); //attributes already in batch
+        private final Batch batch = new Batch();
```

Comment:
Wouldn't it be possible to use rather linked list or some priority queue to hold the dependencies? I believe it would make the code cleaner.